stagemonitor-core/src/main/java/org/stagemonitor/core/elasticsearch/ElasticsearchClient.java

```diff
@@ -547,15 +555,24 @@ public void run() {
 			if (elasticsearchUrl == null) {
 				return;
 			}
-			httpClient.send(HttpRequestBuilder.<Void>forUrl(elasticsearchUrl + "/")
-					.method("HEAD")
+
+			httpClient.send(HttpRequestBuilder.<Void>forUrl(elasticsearchUrl + "/_cluster/health")
+					.method("GET")
 					.successHandler(new HttpClient.ResponseHandler<Void>() {
 						@Override
-						public Void handleResponse(HttpRequest<?> httpRequest, InputStream is, Integer statusCode, IOException e) throws IOException {
-							if (!isElasticsearchAvailable()) {
-								logger.info("Elasticsearch is available again.");
+						public Void handleResponse(HttpRequest<?> httpRequest, InputStream inputStream, Integer statusCode, IOException e) throws IOException {
+							JsonNode clusterHealthResponse = JsonUtils.getMapper().readTree(inputStream);
+							String statusValue = clusterHealthResponse.has("status") ? clusterHealthResponse.get("status").asText() : "red";
+							boolean isNowAvailable = statusValue.equals("green") || statusValue.equals("yellow");
+							if (isNowAvailable) {
+								if (!isElasticsearchAvailable()) {
+									logger.info("Elasticsearch is available again.");
+								}
+								elasticsearchAvailable.set(true);
```

Comment:
Set available to true only after the observers have been notified, otherwise the index pattern might not have been initialized when spans are sent to Elasticsearch